### PR TITLE
feat(core): add architecture/route-component-import

### PR DESCRIPTION
**Pull request by oekazuma**
## Why

A SvelteKit route entry — `+page.svelte`, `+layout.svelte`, `+error.svelte` — is written on the assumption that Kit renders it. Kit hands a page its `data` and `params`; it hands an error page its `page.error` and `page.status`. Imported from somewhere else, the component receives none of that and renders against nothing, or against the importing page's data standing in for its own.

The mistake is easy to make and reads as reasonable: another page needs the same markup, the markup already exists in a `+page.svelte`, so it gets imported. Nothing in the toolchain objects. The component renders — emptily.

## What

`architecture/route-component-import`, at `info`. It resolves each import specifier to a project-relative path, and reports one whose basename is a route-entry name **and** which sits under the routes directory. Those two conditions are separate on purpose: a file named `+page.svelte` outside `src/routes` is not a route entry, because Kit gives those names meaning only there.

**This is the first Architecture rule that is on by default.** The three that shipped before it assert nothing until configured, so a design error there costs a user nothing; here it reaches everyone who upgrades. Two consequences shaped the design.

**The route-entry matcher reproduces Kit's own, not a tidier approximation.** `analyze()` in `@sveltejs/kit/src/core/sync/create_manifest_data/index.js` strips only the component extension before testing `/^\+(?:(page(?:@(.*))?)|(layout(?:@(.*))?)|(error))$/`, so the `@` breakout suffix is unbounded — a layout name may contain dots, and `+page@foo.bar.svelte` is a real route entry. A narrower `[^./]*` would silently skip it.

**`exemptImporters` ships deliberately narrow.** A `string-list` option adds to its default and can never shrink it, so the two failure directions are not symmetric:

| If the default is | The failure is | Can a user fix it? |
| --- | --- | --- |
| too narrow | a false positive | **yes** — append to `exemptImporters` |
| too broad | a missed true positive | **no** — a `string-list` cannot be shrunk |

Only the narrow side leaves a lever, which decides it. Stories, tests and specs are exempt out of the box; configuring the option is an expected step for a project whose satellite convention is its own, and the rule page says so rather than treating it as an edge case.

An exempt importer earns a **pass**, not silence — its route-entry imports really are fine, which is a true statement worth recording. Putting the exemption into `applies` instead would call such a file signal-free, which it is not.

## Supporting changes

- **`ComponentFacts.importSpans` gains `type?: true`** — an import that contributes no runtime value binding. It covers `import type …` and a declaration whose every specifier is inline-typed; a specifier-less side-effect import stays unmarked, because the module really is loaded. Without it the rule would report `import type P from './+page.svelte'`, which renders nothing.
- **`componentRule` hands `applies`/`bad` the `RuleContext`**, which its sibling `kitModuleRule` already did. The rule needs `ctx.project.kitAliases` to resolve a specifier through a project's declared aliases (#330); without it the rule would be blind to exactly the imports it was measured to need. Purely additive — all 23 existing callers declare fewer parameters and are unaffected.

## Measured reach, recorded rather than assumed

The design was measured against a real monorepo of several SvelteKit apps on a convention-compliant branch. **No route entry was imported anywhere, by any file type** — the search pattern was validated first against a synthetic file carrying a relative import, an alias import with an `@` breakout, and a type-only import, all three of which matched. So the rule reports nothing and misses nothing on that tree. The built-in exempt list also covered only a minority of that tree's satellite files, which is why configuration is documented as expected rather than exceptional.

## Not reported

Dynamic `import()` (not an import declaration); an import made from a plain `.ts`/`.js` file, or from a `.svelte.ts`/`.svelte.js` runes module (the parser leaves those files' import spans empty); a type-only import; and a project whose routes live outside `src/routes`. The first is a genuine gap; the second is load-bearing — it is why the exempt list can be three entries long instead of an open-ended guess at every project's test-file convention.

## Verification

`core` 1119, `cli` 779, `vite` 205, `mcp` 25 tests pass; typecheck clean in all four packages; lint and format clean. Six independent mutations are each caught by a test — including the dotted `@` suffix, the type-only skip, and moving the exemption into `applies` — and one test carries a route entry from real source text through the parser into the rule, so the seam between the fact and its consumer is covered rather than assumed.

Design: `docs/superpowers/specs/2026-07-30-route-component-import-design.md`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added an architecture rule that flags Svelte components importing SvelteKit route entries.
  * Includes built-in exemptions for stories, tests, and specs, with configurable additional exemptions.
  * Type-only imports and non-route imports are ignored.

* **Documentation**
  * Added English and Japanese documentation, configuration guidance, examples, and rule listings.
  * Added release notes for the new rule.

* **Tests**
  * Added coverage for route detection, aliases, exemptions, and import classifications.

### CODE REVIEW

File: docs/src/content/docs/rules/index.mdx [hunk 1]
```diff
@@ -24,7 +24,7 @@ The severities below are the defaults — see [Configuration](/guides/configurat
   </Card>
   <Card title="Architecture" icon="layers" href="/rules/architecture">
     Component size, prop surface, and the import boundaries a project declares — signals that code is placed or shaped
-    wrong. (6 rules)
+    wrong. (7 rules)
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply docs/src/content/docs/rules/index.mdx]: Fixed in 5fd3c298 — but the stated reason does not hold, and the real fix was somewhere else.

These counts were **already generated**: `packages/cli/scripts/rules-index.mjs` had a `RULE_COUNT` map and a `withRuleCount()` that appended the suffix to each category blurb, inside the block `gen-rules-index.mjs` rewrites. So they never went stale — `packages/cli/test/rules-index.test.mjs` fails the build if the generated block is out of date, and this PR's bump from 6 to 7 came out of the generator, not a hand edit.

Editing only the two `.mdx` files, as the comment proposes, would therefore have been undone by the next `gen:rules-index` run and would have failed that test in the meantime.

The change is still right, for the convention rather than for rot. `AGENTS.md` states it verbatim: "Never hard-code rule counts or ID ranges in READMEs/guides … refer to rule *categories* instead." A generated count also churns five card lines on every rule addition for no reader benefit. So `RULE_COUNT`/`withRuleCount` are gone from the generator, the block is regenerated, and the counts are removed from **every** category card in both languages — fixing only Architecture while leaving four others would have been arbitrary.

File: docs/superpowers/plans/2026-07-31-route-component-import.md [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+## Not reported
+
+- A dynamic `import()` of a route entry — it is not an import declaration, so the analyzer does not see it.
+- An import made from a plain `.ts` or `.js` file, or from a `.svelte.ts` / `.svelte.js` module: import
+  facts are collected from `.svelte` component files only.
+- A type-only import (`import type P from './+page.svelte'`, or one whose every specifier is inline-typed)
+  — it is erased at build, so nothing renders.
+- A project whose routes live somewhere other than `src/routes`.
+
+````
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply docs/superpowers/plans/2026-07-31-route-component-import.md]: Fixed in 5fd3c298. The four-backtick fence closed before the example's own "## Not reported" section, so that section escaped into the plan's prose and the remaining fence opened an unlabeled block. The outer fence now closes after the section, and one nearby asymmetric fence (three-backtick open against a four-backtick close) is normalised too. All fences in the file balance.

File: packages/core/src/rules/architecture/route-component-import.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * be collected rather than pinned for the process lifetime.
+ */
+const routeEntryImportsCache = new WeakMap<ComponentFacts, { line: number; target: string }[]>();
+
+function cachedRouteEntryImports(c: ComponentFacts, ctx: RuleContext): { line: number; target: string }[] {
+  const cached = routeEntryImportsCache.get(c);
+  if (cached !== undefined) return cached;
+  const result = routeEntryImports(c, ctx);
+  routeEntryImportsCache.set(c, result);
+  return result;
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply packages/core/src/rules/architecture/route-component-import.ts:78]: Correct, and fixed in 5fd3c298 — though not by removing the cache.

The cache exists because `componentRule` calls `applies` and `bad` back to back for the same component; without it every import specifier in the project is resolved twice, which a final-review pass measured at ~27-30% of this rule's cost. This rule is on by default, so that is paid on every dev-server save. Removing it trades a real, universal cost for a hypothetical one.

Keying it correctly costs nothing and closes the hole exactly. `routeEntryImports` reads three things: `c.importSpans`, `c.file`, and `ctx.project.kitAliases`. The first two *are* the `ComponentFacts` identity; only the third was missing. The entry now carries the aliases reference it was computed under:

```ts
const cache = new WeakMap<ComponentFacts, { aliases: Project['kitAliases']; result: … }>();
```

and a differing reference recomputes. Reference equality is the right comparison here: the alias list is rebuilt per collection, so a new analysis always brings a new array, and two structurally-equal-but-distinct arrays merely miss the cache — a recomputation, never a wrong answer.

Worth noting why this was reachable in the first place. Nothing in this repo's own pipeline reuses `ComponentFacts` across analyses (`collectComponentFacts` reparses every time), so the bug had no in-repo trigger. But `packages/core` is a published library and `RuleContext` is part of its surface, so that invariant was never this module's to assume — which is exactly the finding.

A test now runs the rule twice over the **same** `ComponentFacts` object with two different alias lists, where the specifier resolves to a route entry under one and not the other, and asserts each call answers for its own aliases. It was confirmed failing before the fix (the second call returned the first call's result).

File: packages/core/src/rules/architecture/route-component-import.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  severity: 'info',
+  label: 'Route component imports',
+  options: { exemptImporters: { kind: 'string-list', default: EXEMPT_IMPORTERS } },
+  recommendation:
+    'Extract the shared markup into a component under $lib and import that from both places, leaving the route entry to SvelteKit.',
+  rationale:
+    'A route entry is written on the assumption that SvelteKit renders it: Kit hands a page its data and params, and an error page its page.error and page.status. Imported from somewhere else it receives none of that and renders against nothing, or against the importing page data standing in for its own.',
+  // Signal present = this file imports a route entry, exempt or not. An exempt file therefore
+  // reaches `bad` and earns a PASS, rather than being called signal-free.
+  applies: (c, o, ctx) => cachedRouteEntryImports(c, ctx).length > 0,
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply packages/core/src/rules/architecture/route-component-import.ts]: Renamed to `_o` in f83ac88c, with a comment saying why the parameter is there at all.

The lint half of this does not hold, though, and it is worth recording so nobody re-litigates it: `oxlint .` is clean on this file today, both before and after the rename. `no-unused-vars` defaults to `args: "after-used"`, so a parameter followed by a used one is never reported — `o` sits before `ctx`, which is used. The rename changes nothing for the linter.

The readability half is fair. Every other component rule simply omits what it does not use (`applies: (c) => …`), which is not available here because `ctx` is third — so the naked `o` reads like an oversight rather than a positional placeholder. `_o` plus one comment says which it is.

Note that this is the first `_`-prefixed parameter in `packages/core/src`; there was no existing convention to follow, because this is the first rule that needed `ctx` and therefore the first with a middle parameter to skip.
